src/resolve.rs

```diff
@@ -0,0 +1,49 @@
+use tokio::net::lookup_host;
+use url::Url;
+
+/// Resolve a URL to a socket address
+pub async fn resolve_address_from_url(url: &str) -> std::io::Result<std::net::SocketAddr> {
+    let url = Url::parse(url).map_err(|e| std::io::Error::other(format!("Invalid URL: {}", e)))?;
+
+    let host = url
+        .host_str()
+        .ok_or_else(|| std::io::Error::other("Missing host"))?;
+    let port = url
+        .port()
+        .ok_or_else(|| std::io::Error::other("Missing port"))?;
+
+    let maybe_addr = lookup_host(format!("{host}:{port}")).await?.next();
```

Comment:
This is what `ToSocketAddrs` calls. So if somebody uses `ws://localhost:8888` rather than `ws://127.0.0.1:8888`, we won't crash and burn